babblesim/hw/mcu/nordic/nrf52_bsim/src/nrf52_clock.c

```diff
@@ -54,9 +55,7 @@ nrf52_clock_hfxo_request(void)
              CLOCK_HFCLKSTAT_STATE_Running << CLOCK_HFCLKSTAT_STATE_Pos)) {
             NRF_CLOCK->EVENTS_HFCLKSTARTED = 0;
             nrf_clock_task_trigger(NRF_CLOCK, NRF_CLOCK_TASK_HFCLKSTART);
-            while (!NRF_CLOCK->EVENTS_HFCLKSTARTED) {
-                tm_tick();
-            }
+            BUSY_LOOP (!NRF_CLOCK->EVENTS_HFCLKSTARTED)
```

Comment:
tbh I much more prefer `#ifdef BABBLESIM` (or whatever can be used to identify BabbleSim) to call `tm_tick()` that using such macros. alternatively, the code can be rewritten to use `wfi` instead of busy loop, but I am not sure if this can be done in all cases.